datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -944,15 +944,24 @@ private void uninitializeTasks(List<DatastreamTask> tasks) {
     tasks.forEach(_cpProvider::unassignDatastreamTask);
   }
 
-  private void initializeTask(DatastreamTask task, Set<DatastreamTask> failedDatastreamTasks, boolean retryAndSaveError) {
+  private void initializeTask(DatastreamTask task, Set<DatastreamTask> failedDatastreamTasks,
+      boolean retryAndSaveError) {
     try {
       DatastreamTaskImpl taskImpl = (DatastreamTaskImpl) task;
       assignSerdes(taskImpl);
 
       boolean customCheckpointing = getCustomCheckpointing(task);
       TransportProviderAdmin tpAdmin = _transportProviderAdmins.get(task.getTransportProviderName());
       TransportProvider transportProvider = tpAdmin.assignTransportProvider(task);
-      EventProducer producer = new EventProducer(task, transportProvider, _cpProvider, _eventProducerConfig, customCheckpointing);
+
+      Function<DatastreamTask, Set<String>> throughputViolatingTopicsProvider =
+          _config.getEnableThroughputViolatingTopicsHandling() ?
+              (t) -> _adapter.getThroughputViolatingTopics(t.getDatastreams()) :
+              (t) -> new HashSet<>();
```

Comment:
Will it make sense to simplify this at Datastream level rather than DatastreamTask level?

For eg: in digest, there is only one datastream. This function will result in 10 copies for the same datastream.